lib/editor/components/BulkEditorModal.js

```diff
@@ -0,0 +1,396 @@
+            onChange={this.handleAddField}
+          >
+            <option value=''>Select field to update</option>
+            {
+              currentTable.fields
+                .filter(field => field.bulkEditEnabled)
+                .map((field, i) => (<option key={field.name}>{field.name}</option>))
+            }
+          </FormControl>
+          {patch
+            ? Object.keys(patch).map(key => {
+              const field = currentTable.fields.find(f => f.name === key)
+              if (!field) throw new Error(`Cannot find field for key: ${key}`)
+              return (
+                <PatchField
+                  key={field.name}
+                  activeComponent={activeComponent}
+                  approveGtfsDisabled={this.props.approveGtfsDisabled}
+                  onChange={this.handleChangeField}
+                  onRemove={this.handleRemoveField}
+                  patch={patch || {}}
+                  field={field}
+                  tableData={tableData}
+                />
+              )
+            })
+            : null
+          }
+          <hr />
+          <ControlLabel>Add field to filter on:</ControlLabel>
+          <FormControl
+            componentClass='select'
+            value=''
+            style={{ display: 'inline', width: '200px', marginLeft: '10px' }}
+            onChange={this.handleAddFilterField}
+            onRemove={this.handleRemoveFilterField}
+          >
+            <option value=''>-- select a field to filter on --</option>
+            {
+              currentTable.fields
+                .map((field, i) => (<option key={field.name}>{field.name}</option>))
+            }
+          </FormControl>
+          <div
+            className='text-center lead'
+            style={{ margin: '10px' }}>
+            Changes apply to {hasFilters ? 'records where:' : 'all records.'}
+          </div>
+          {filter
+            ? Object.keys(filter).map(key => {
+              const field = currentTable.fields.find(f => f.name === key)
+              if (!field) throw new Error(`Cannot find field for key: ${key}`)
+              return (
+                <FilterField
+                  key={field.name}
+                  onChange={this.handleChangeFilter}
+                  onRemove={this.handleRemoveFilterField}
+                  filter={filter || {}}
+                  field={field}
+                />
+              )
+            })
+            : null
+          }
+          <Alert bsStyle='warning' style={{ marginTop: '20px' }}>
+            <strong>Warning:</strong> Bulk editing will modify all records (or
+            those records matching the filter), but this operation cannot be undone!
+            <hr />
+            It is recommended to first create a snapshot of the current state of
+            the data in the editor before applying a bulk edit. A snapshot takes
+            only a few seconds and allows you to revert your changes if the bulk
```

Comment:
Minor language quibble: I think it'd be more appropriate to say it is possible to "restore" instead of "revert", but maybe it's not such a huge difference.

Reply:
Yea, I think this is relatively minor. I think the language of `revert your changes` is a bit easier to digest than `restore the previous state before the bulk edit`.